# DAP019

SQL parameters were detected, but no parameters are being supplied. Example:

``` csharp
conn.Execute("update Customers set Name=@name where Id=@id"); // note no args supplied
```

How reliable this message is depends on whether the tooling knows your SQL variant. Currently, only TSQL (SQL Server)
has full / reliable SQL parsing; without that, which can help avoid false-positives from local
variables and table-variables:

``` csharp
conn.Execute("""
    declare @id int, @name nvarchar(200);
    -- assignment not shown
    update Customers set Name=@name where Id=@id");
```

Here we don't need any arguments, as `@id` and `@name` are not actually parameters, but have been incorrectly detected
by the basic parser.

There are 3 ways to avoid this:

1. if the connection is statically known to be `SqlConnection`, it will assume TSQL
2. if Dapper.AOT is installed and `[SqlSyntax(SqlSyntax.SqlServer)]` is in scope, it will assume TSQL
3. if the `<Dapper_SqlSyntax>SqlServer</Dapper_SqlSyntax>` property is specified in the project file (in a `<PropertyGroup>`), it will assume TSQL