# DAP051

TypeHandler attribute points to a type handler class (TTypeHandler in TypeHandler<TValue, TTypeHandler>) that is not a valid named type,
or cannot be constructed as required by Dapper AOT.
The type handler must be a concrete (non-abstract) class and must have a public parameterless constructor.

Bad:

``` c#
[module: TypeHandler<MyClass, IMyHandler>]
[module: TypeHandler<MyClass, MyAbstractHandler>]
```

Good:

``` c#
[module: TypeHandler<MyClass, MyHandler>]
```